### PR TITLE
feat(Gate-Validation): add endpoints to validate whether record can e…

### PR DESCRIPTION
**Pull request by nicoprow**
Sharing members should have a way to determine whether a record can enter the sharing process

### CODE REVIEW

File: bpdm-gate/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/controller/SiteController.kt [hunk 1]
```diff
@@ -112,4 +115,21 @@ class SiteController(
...
     }
+
+    @Operation(
+        summary = "Validate a site",
+        description = "Determines errors in a site record which keep it from entering the sharing process"
+    )
+    @ApiResponses(
+        value = [
+            ApiResponse(responseCode = "200", description = "A validation response with possible errors"),
+            ApiResponse(responseCode = "400", description = "On malformed legal entity requests", content = [Content()]),
```
Comment (pazepaze) [inline bpdm-gate/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/controller/SiteController.kt:126]: legal entity -> site

File: bpdm-gate/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/controller/AddressController.kt [hunk 2]
```diff
@@ -117,4 +120,21 @@ class AddressController(
...
     }
+
+    @Operation(
+        summary = "Validate an address partner",
+        description = "Determines errors in an address partner record which keep it from entering the sharing process"
+    )
+    @ApiResponses(
+        value = [
+            ApiResponse(responseCode = "200", description = "A validation response with possible errors"),
+            ApiResponse(responseCode = "400", description = "On malformed legal entity requests", content = [Content()]),
```
Comment (pazepaze) [inline bpdm-gate/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/controller/AddressController.kt:131]: legal entity -> address

**Review by pazepaze — CHANGES_REQUESTED** [review]

File: bpdm-gate/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/controller/SiteController.kt (hunk 1 above)
Comment (nicoprow) [reply bpdm-gate/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/controller/SiteController.kt:126]: done

File: bpdm-gate/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/controller/AddressController.kt (hunk 2 above)
Comment (nicoprow) [reply bpdm-gate/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/controller/AddressController.kt]: done

File: bpdm-gate/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/controller/LegalEntityController.kt [hunk 3]
```diff
@@ -113,4 +116,22 @@ class LegalEntityController(
...
     }
+
+    @Operation(
+        summary = "Validate a legal entity",
+        description = "Determines errors in a legal entity record which keep it from entering the sharing process"
+    )
+    @ApiResponses(
+        value = [
+            ApiResponse(responseCode = "200", description = "A validation response with possible errors"),
+            ApiResponse(responseCode = "400", description = "On malformed site requests", content = [Content()]),
```
Comment (pazepaze) [inline bpdm-gate/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/controller/LegalEntityController.kt:127]: Should be "legal entity" here and "site" in the SiteController

File: bpdm-gate/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/controller/SiteController.kt (hunk 1 above)
Comment (pazepaze) [reply bpdm-gate/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/controller/SiteController.kt:126]: should be "site" here

**Review by pazepaze — CHANGES_REQUESTED** [review]

**Review by pazepaze — APPROVED** [review]

**Pull request merged**